emission/analysis/userinput/matcher.py

```diff
@@ -61,34 +62,105 @@ def handle_single_most_recent_match(confirmed_trip, ui):
+    if "additions" not in confirmed_object["data"] or \
+        confirmed_object["data"]["additions"] is None:
+        confirmed_object["data"]["additions"] = []
     if "status" not in ui.data or ui.data.status == ecwtui.InputStatus.ACTIVE:
-        confirmed_trip["data"]["trip_addition"].append(ui)
+        confirmed_object["data"]["additions"].append(ui)
     elif ui.data.status == ecwtui.InputStatus.DELETED:
-        after_del_list = [ta for ta in confirmed_trip["data"]["trip_addition"] if ta["match_id"] != ui["match_id"]]
-        confirmed_trip["data"]["trip_addition"] = after_del_list
+        after_del_list = [ta for ta in confirmed_object["data"]["additions"] if ta["match_id"] != ui["match_id"]]
+        confirmed_object["data"]["additions"] = after_del_list
     else:
         # TODO: Decide whether to error or to warn here
         logging.warn("Invalid status found in user input %s, moving forward anyway" % ui)
 
+def create_composite_objects(user_id):
+    time_query = epq.get_time_range_for_composite_object_creation(user_id)
+    ts = esta.TimeSeries.get_time_series(user_id)
+    confirmedTrips = esda.get_entries(esda.CONFIRMED_TRIP_KEY, user_id, time_query=time_query)
+    logging.info("Creating composite trips from %d confirmed trips" % len(confirmedTrips))
+    if len(confirmedTrips) == 0:
+        logging.debug("len(confirmedTrips) == 0, early return")
+        return None
+    for ct in confirmedTrips:
+        logging.info("End place type for trip is %s" %  type(ct.data.end_place))
+        composite_trip_dict = copy.copy(ct)
+        del composite_trip_dict["_id"]
+        composite_trip_dict["metadata"]["key"] = "analysis/composite_trip"
+
+        # confirmed_trip has an id for its corresponding confirmed_place
+        # for composite_trip, we want to get the actual confirmed_place object
+        confirmed_place_id = ct["data"]["confirmed_place"]
+        confirmed_place = esda.get_entry(esda.CONFIRMED_PLACE_KEY, confirmed_place_id)
+        composite_trip_dict["data"]["confirmed_place"] = confirmed_place
+
+        # retrieve locations for the trajectory of the trip
+        time_query = estt.TimeQuery("data.ts", ct["data"]["start_ts"], ct["data"]["end_ts"])
+        locations = esda.get_entries(esda.CLEANED_LOCATION_KEY, user_id, time_query=time_query)
+        max_entries = 100; # we will downsample to 100 locations
+        if len(locations) > max_entries:
+            logging.debug('Downsampling to %d points' % max_entries)
+            sample_rate = len(locations)//max_entries + 1
+            locations = locations[::sample_rate]
+        composite_trip_dict["data"]["locations"] = locations
+
+        # later we will want to put section & modes in composite_trip as well
+
+        composite_trip_entry = ecwe.Entry(composite_trip_dict)
+        # save the entry
+        ts.insert(composite_trip_entry)
+    epq.mark_composite_object_creation_done(user_id, None)
```

Comment:
When marking the composite creation stage done, we should pass the timestamp of the last place instead of "None"